fix(kuma-cp): retrieve name from owner not parsing pod name for Deployments/CronJob (#5569)
kumahq/kuma: UPGRADE.md

## Upcoming release

### **List of breaking changes**

> #### **Naming of Serviceless has changed**

Reviewer: ```suggestion
### **Naming Serviceless dataplanes has changed**
```

Reviewer 2: ```suggestion
#### **Naming of services for `Pods` without a `Service` has changed**
```

> Currently, serviceless pods' name is generated based on the `Pod` name. `Kuma-cp` takes the pod's name and removes 2 last elements after splitting by `-`. This behavior is correct when `Pod` is managed by `Deployment` or `CronJob` but not for other types. We've introduced the change that takes the name from the parent resource which breaks current behavior.

Reviewer 2: ```suggestion
Currently, the `kuma.io/service` value of the inbound of a `Dataplane` generated for a `Service`less `Pod` is based on the `Pod` name. The Kuma CP takes the pod's name and removes 2 last elements after splitting by `-`. This behavior is correct when the `Pod` is owned by a `Deployment` or `CronJob` but not for other owner kinds. Kuma will now use the name of the owner resource as the `kuma.io/service` value.
```

Reviewer 2: we need to be deliberate with how we explain this. a service is not a `Service`.

Before upgrade:
1. Identify all serviceless pods that are not managed by `Deployment` or `CronJob`.
2. Create policies that were created for services but with a new name. The name is an owner name, in case there is no owner `Kuma` uses `Pod` name.

> 3. Upgrade `Kuma`.

Reviewer: Need to explain why it's ok to do it as a breaking change

> 1. Identify all serviceless pods that are not managed by `Deployment` or `CronJob`.
> 2. Create policies that were created for services but with a new name. The name is an owner name, in case there is no owner `Kuma` uses `Pod` name.
> 3. Upgrade `Kuma`.

Reviewer 2: ```suggestion
1. Identify all `Service`less `Pods` that are not managed by a `Deployment` or `CronJob`.
2. Create copies of policies that were created for the services corresponding to these `Pods`. The `kuma.io/service` value is the name of the owner resource. If there is no owner, `Kuma` uses the `Pod`'s name.
```

> ### Other changes

Reviewer: ```suggestion
```

> #### http api

Reviewer: ```suggestion
### http api
```

We've removed the endpoint `POST /tokens` which has been moved by `POST /tokens/dataplane` (same request and response) some time ago.
Make sure you are using a recent `kumactl` or that you use the right path if using the api directly to upgrade with no issues.

#### Kubernetes

The sidecar container is always injected first (since [#5436](https://github.com/kumahq/kuma/pull/5436)). This should only impact you when modifying the sidecar container with a container-patch. If you do so, upgrade Kuma and then change your container patch to modify the right container.

> #### Kumactl

Reviewer: ```suggestion
### Kumactl
```
